src/solver/ts-generator/optimized-thermal-generator/optimization/SetVariableBounds.cpp

```diff
@@ -0,0 +1,96 @@
+//
+// Created by milos on 14/11/23.
+//
+
+#include "../main/OptimizedGenerator.h"
+
+namespace Antares::Solver::TSGenerator
+{
+
+// TODO CR27: see if to make this bound or constraint -
+// it is definitely easier to do set it as a fix bound -
+// but the solver might go crazy - as for adq.patch
+
+// this will fix some start & end variable bounds to 0 or 1
+void OptimizedThermalGenerator::fixBounds()
+{
+    // loop per units
+    for (const auto& unit : vars.clusterUnits)
+    {
+        fixBounds(unit);
+    }
+    return;
+}
+
+void OptimizedThermalGenerator::fixBounds(const Unit& unit)
+{
+    if (!unit.createStartEndVariables)
+        return;
+
+    int averageMaintenanceDuration = par.getAverageMaintenanceDuration(*(unit.parentCluster));
+
+    fixBoundsFirstMnt(unit);
+    fixBounds(unit, averageMaintenanceDuration);
+}
+
+// Bounds for the start of the first maintenance
+// first maintenance must start between tauLower and tauUpper
+// start[u][0][tauLower-1] = 0
+// start[u][0][tauUpper] = 1
+void OptimizedThermalGenerator::fixBoundsFirstMnt(const Unit& unit)
+{
+    int earliestStartOfFirstMaintenance
+      = par.calculateUnitEarliestStartOfFirstMaintenance(*(unit.parentCluster), unit.index);
+    int latestStartOfFirstMaintenance
+      = par.calculateUnitLatestStartOfFirstMaintenance(*(unit.parentCluster), unit.index);
+
+    //
+    // We assume here that vector "maintenance" has member [0]
+    // meaning: for each unit we assume we have at least one maintenance
+    // this assumption is ok - since method calculateNumberOfMaintenances()
+    // will never return number bellow 2
+
+    if (earliestStartOfFirstMaintenance >= 1)
+    {
+        // start[u][0][tauLower-1] = 0
+        unit.maintenances[0].start[earliestStartOfFirstMaintenance - 1]->SetBounds(0.0, 0.0);
+    }
+
+    // start[u][0][tauUpper] = 1
+    unit.maintenances[0].start[latestStartOfFirstMaintenance]->SetBounds(1.0, 1.0);
```

Comment:
Should be `unit.maintenances[0].start[latestStartOfFirstMaintenance - 1]->SetBounds(1.0, 1.0);` because arrays starts from 0.